lib/backend/dynamo/dynamodbbk.go

```diff
@@ -562,6 +572,47 @@ func (b *Backend) Delete(ctx context.Context, key []byte) error {
 	return b.deleteKey(ctx, key)
 }
 
+// ConditionalUpdate updates the matching item in Dynamo if the provided revision matches
+// the revision of the item in Dynamo.
+func (b *Backend) ConditionalUpdate(ctx context.Context, item backend.Item) (*backend.Lease, error) {
+	rev, err := b.create(ctx, item, modeConditionalUpdate)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	item.Revision = rev
+	return backend.NewLease(item), nil
+}
+
+// ConditionalDelete deletes item by key if the provided revision matches
+// the revision of the item in Dynamo.
+func (b *Backend) ConditionalDelete(ctx context.Context, key []byte, rev string) error {
+	if len(key) == 0 {
+		return trace.BadParameter("missing parameter key")
+	}
```

Comment:
`create` (as called by `Create`, `Update` and `Put` and now `ConditionalUpdate`) doesn't check that the key is nonempty.

Reply:
If anything we should allow deleting a key that's empty but not write it - right now we're in the weird position of allowing a Put with an empty key with no way to delete it 😬 

If we can distinguish between an empty but set FullPath and no FullPath (as we use that for existence), why not just allow the empty key?